src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/MySqlEngine.java

```diff
@@ -861,14 +863,25 @@ protected void setSchema(final String schema) throws DatabaseEngineException {
     }
 
     @Override
-    public synchronized ResultIterator iterator(String query) throws DatabaseEngineException {
+    public ResultIterator iterator(String query) throws DatabaseEngineException {
+        return iterator(query, properties.getFetchSize(), properties.getSelectQueryTimeout());
+    }
```

Comment:
I suppose you kept this one because the super method has a default fetch size of DEFAULT_FETCH_SIZE instead of `properties.getFetchSize()`?
Turns out that didn't seem very correct, and was fixed.
If you rebase to the head of "master" branch, I think calling the `super.iterator(String query)` will end up using the same fetch size and select query timeout as this method, so the override won't be needed anymore.